ERROR: gcc-sanitizers-8.3.0-r0 do_compile: oe_runmake failed

when trying to add EXTRA_IMAGE_FEATURES ?= "debug-tweaks tools-sdk dev-pkgs" so there was a compiler and dev tools in the image it failed with

ERROR: gcc-sanitizers-8.3.0-r0 do_compile: oe_runmake failed
ERROR: gcc-sanitizers-8.3.0-r0 do_compile: Execution of '/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/run.do_compile.3850804' failed with exit code 1:
make "AR_FLAGS=" "CC_FOR_BUILD=gcc " "CFLAGS=-O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot-native= " "CXXFLAGS=-O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot-native=  -fvisibility-inlines-hidden" "CFLAGS_FOR_BUILD=-isystem/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot-native/usr/include -O2 -pipe" "CFLAGS_FOR_TARGET=" "INSTALL=/home/dingo/tegra-demo-distro/build/tmp/hosttools/install -c" "INSTALL_DATA=/home/dingo/tegra-demo-distro/build/tmp/hosttools/install -c -m 644" "INSTALL_PROGRAM=/home/dingo/tegra-demo-distro/build/tmp/hosttools/install -c" "INSTALL_SCRIPT=/home/dingo/tegra-demo-distro/build/tmp/hosttools/install -c" "JC1FLAGS=" "LDFLAGS=-Wl,-O1 -Wl,--hash-style=gnu -Wl,--as-needed -Wl,-z,relro,-z,now" "LIBCFLAGS=" "LIBCFLAGS_FOR_TARGET=" "MAKE=make" "MAKEINFO=/bin/sh /home/dingo/tegra-demo-distro/build/tmp/work-shared/gcc-8.3.0-r0/gcc-8.3.0/missing --run makeinfo " "PICFLAG=" "PICFLAG_FOR_TARGET=" "SHELL=/bin/sh" "RUNTESTFLAGS=" "exec_prefix=/usr" "infodir=/usr/share/info" "libdir=/usr/lib" "prefix=/usr" "includedir=/usr/include" "AR=aarch64-oe4t-linux-gcc-ar" "AS=aarch64-oe4t-linux-as " "LD=aarch64-oe4t-linux-ld --sysroot=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot " "LIBCFLAGS=" "NM=aarch64-oe4t-linux-gcc-nm" "PICFLAG=" "RANLIB=aarch64-oe4t-linux-gcc-ranlib" "DESTDIR=" all-recursive
make[1]: Entering directory '/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/gcc-8.3.0/build.aarch64-oe4t-linux.aarch64-oe4t-linux/aarch64-oe4t-linux/libsanitizer'
Making all in sanitizer_common
make[2]: Entering directory '/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/gcc-8.3.0/build.aarch64-oe4t-linux.aarch64-oe4t-linux/aarch64-oe4t-linux/libsanitizer/sanitizer_common'
/bin/sh ../libtool --tag=CXX   --mode=compile aarch64-oe4t-linux-g++  -march=armv8-a+crc -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security -Wdate-time --sysroot=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot -D_GNU_SOURCE -D_DEBUG -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS  -DHAVE_RPC_XDR_H=0 -DHAVE_TIRPC_RPC_XDR_H=0 -I. -I../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/sanitizer_common -I..  -I ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/include -isystem ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/include/system  -Wall -W -Wno-unused-parameter -Wwrite-strings -pedantic -Wno-long-long -fPIC -fno-builtin -fno-exceptions -fno-rtti -fomit-frame-pointer -funwind-tables -fvisibility=hidden -Wno-variadic-macros  -std=gnu++11  -DSANITIZER_LIBBACKTRACE -DSANITIZER_CP_DEMANGLE -I ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/../libbacktrace -I ../libbacktrace -I ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/../include -include ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/libbacktrace/backtrace-rename.h -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot-native=  -fvisibility-inlines-hidden -c -o sanitizer_platform_limits_posix.lo ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc
libtool: compile:  aarch64-oe4t-linux-g++ -march=armv8-a+crc -fstack-protector-strong -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security -Wdate-time --sysroot=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot -D_GNU_SOURCE -D_DEBUG -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -DHAVE_RPC_XDR_H=0 -DHAVE_TIRPC_RPC_XDR_H=0 -I. -I../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/sanitizer_common -I.. -I ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/include -isystem ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/include/system -Wall -W -Wno-unused-parameter -Wwrite-strings -pedantic -Wno-long-long -fPIC -fno-builtin -fno-exceptions -fno-rtti -fomit-frame-pointer -funwind-tables -fvisibility=hidden -Wno-variadic-macros -std=gnu++11 -DSANITIZER_LIBBACKTRACE -DSANITIZER_CP_DEMANGLE -I ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/../libbacktrace -I ../libbacktrace -I ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/../include -include ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/libbacktrace/backtrace-rename.h -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0 -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0 -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot= -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot-native= -fvisibility-inlines-hidden -c ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc  -fPIC -DPIC -o .libs/sanitizer_platform_limits_posix.o
../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc:160:10: fatal error: linux/cyclades.h: No such file or directory
 #include <linux/cyclades.h>
          ^~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [Makefile:541: sanitizer_platform_limits_posix.lo] Error 1
make[2]: Leaving directory '/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/gcc-8.3.0/build.aarch64-oe4t-linux.aarch64-oe4t-linux/aarch64-oe4t-linux/libsanitizer/sanitizer_common'
make[1]: *** [Makefile:472: all-recursive] Error 1
make[1]: Leaving directory '/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/gcc-8.3.0/build.aarch64-oe4t-linux.aarch64-oe4t-linux/aarch64-oe4t-linux/libsanitizer'
make: *** [Makefile:342: all] Error 2
WARNING: /home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/run.do_compile.3850804:193 exit 1 from 'exit 1'
WARNING: Backtrace (BB generated script): 
	OE4T/tegra-demo-distro#1: bbfatal_log, /home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/run.do_compile.3850804, line 193
	OE4T/tegra-demo-distro#2: die, /home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/run.do_compile.3850804, line 174
	OE4T/tegra-demo-distro#3: oe_runmake, /home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/run.do_compile.3850804, line 163
	OE4T/tegra-demo-distro#4: do_compile, /home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/run.do_compile.3850804, line 158
	OE4T/tegra-demo-distro#5: main, /home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/run.do_compile.3850804, line 197

Backtrace (metadata-relative locations):
	OE4T/tegra-demo-distro#1: bbfatal_log, /home/dingo/tegra-demo-distro/layers/meta/classes/logging.bbclass, line 72
	OE4T/tegra-demo-distro#2: die, /home/dingo/tegra-demo-distro/layers/meta/classes/base.bbclass, line 56
	OE4T/tegra-demo-distro#3: oe_runmake, /home/dingo/tegra-demo-distro/layers/meta/classes/base.bbclass, line 65
	OE4T/tegra-demo-distro#4: do_compile, /home/dingo/tegra-demo-distro/layers/meta-tegra/contrib/recipes-devtools/gcc-8/gcc-sanitizers.inc, line 32
ERROR: Logfile of failure stored in: /home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/log.do_compile.3850804
Log data follows:
| DEBUG: Executing python function autotools_aclocals
| DEBUG: SITE files ['endian-little', 'bit-64', 'arm-common', 'arm-64', 'common-linux', 'common-glibc', 'aarch64-linux', 'common']
| DEBUG: Python function autotools_aclocals finished
| DEBUG: Executing shell function do_compile
| NOTE: make -j 36 MULTIBUILDTOP=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/gcc-8.3.0/build.aarch64-oe4t-linux.aarch64-oe4t-linux/aarch64-oe4t-linux/libsanitizer/
| make "AR_FLAGS=" "CC_FOR_BUILD=gcc " "CFLAGS=-O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot-native= " "CXXFLAGS=-O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot-native=  -fvisibility-inlines-hidden" "CFLAGS_FOR_BUILD=-isystem/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot-native/usr/include -O2 -pipe" "CFLAGS_FOR_TARGET=" "INSTALL=/home/dingo/tegra-demo-distro/build/tmp/hosttools/install -c" "INSTALL_DATA=/home/dingo/tegra-demo-distro/build/tmp/hosttools/install -c -m 644" "INSTALL_PROGRAM=/home/dingo/tegra-demo-distro/build/tmp/hosttools/install -c" "INSTALL_SCRIPT=/home/dingo/tegra-demo-distro/build/tmp/hosttools/install -c" "JC1FLAGS=" "LDFLAGS=-Wl,-O1 -Wl,--hash-style=gnu -Wl,--as-needed -Wl,-z,relro,-z,now" "LIBCFLAGS=" "LIBCFLAGS_FOR_TARGET=" "MAKE=make" "MAKEINFO=/bin/sh /home/dingo/tegra-demo-distro/build/tmp/work-shared/gcc-8.3.0-r0/gcc-8.3.0/missing --run makeinfo " "PICFLAG=" "PICFLAG_FOR_TARGET=" "SHELL=/bin/sh" "RUNTESTFLAGS=" "exec_prefix=/usr" "infodir=/usr/share/info" "libdir=/usr/lib" "prefix=/usr" "includedir=/usr/include" "AR=aarch64-oe4t-linux-gcc-ar" "AS=aarch64-oe4t-linux-as " "LD=aarch64-oe4t-linux-ld --sysroot=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot " "LIBCFLAGS=" "NM=aarch64-oe4t-linux-gcc-nm" "PICFLAG=" "RANLIB=aarch64-oe4t-linux-gcc-ranlib" "DESTDIR=" all-recursive
| make[1]: Entering directory '/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/gcc-8.3.0/build.aarch64-oe4t-linux.aarch64-oe4t-linux/aarch64-oe4t-linux/libsanitizer'
| Making all in sanitizer_common
| make[2]: Entering directory '/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/gcc-8.3.0/build.aarch64-oe4t-linux.aarch64-oe4t-linux/aarch64-oe4t-linux/libsanitizer/sanitizer_common'
| /bin/sh ../libtool --tag=CXX   --mode=compile aarch64-oe4t-linux-g++  -march=armv8-a+crc -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security -Wdate-time --sysroot=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot -D_GNU_SOURCE -D_DEBUG -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS  -DHAVE_RPC_XDR_H=0 -DHAVE_TIRPC_RPC_XDR_H=0 -I. -I../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/sanitizer_common -I..  -I ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/include -isystem ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/include/system  -Wall -W -Wno-unused-parameter -Wwrite-strings -pedantic -Wno-long-long -fPIC -fno-builtin -fno-exceptions -fno-rtti -fomit-frame-pointer -funwind-tables -fvisibility=hidden -Wno-variadic-macros  -std=gnu++11  -DSANITIZER_LIBBACKTRACE -DSANITIZER_CP_DEMANGLE -I ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/../libbacktrace -I ../libbacktrace -I ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/../include -include ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/libbacktrace/backtrace-rename.h -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot-native=  -fvisibility-inlines-hidden -c -o sanitizer_platform_limits_posix.lo ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc
| libtool: compile:  aarch64-oe4t-linux-g++ -march=armv8-a+crc -fstack-protector-strong -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security -Wdate-time --sysroot=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot -D_GNU_SOURCE -D_DEBUG -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -DHAVE_RPC_XDR_H=0 -DHAVE_TIRPC_RPC_XDR_H=0 -I. -I../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/sanitizer_common -I.. -I ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/include -isystem ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/include/system -Wall -W -Wno-unused-parameter -Wwrite-strings -pedantic -Wno-long-long -fPIC -fno-builtin -fno-exceptions -fno-rtti -fomit-frame-pointer -funwind-tables -fvisibility=hidden -Wno-variadic-macros -std=gnu++11 -DSANITIZER_LIBBACKTRACE -DSANITIZER_CP_DEMANGLE -I ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/../libbacktrace -I ../libbacktrace -I ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/../include -include ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/libbacktrace/backtrace-rename.h -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0 -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0 -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot= -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot-native= -fvisibility-inlines-hidden -c ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc  -fPIC -DPIC -o .libs/sanitizer_platform_limits_posix.o
| ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc:160:10: fatal error: linux/cyclades.h: No such file or directory
|  #include <linux/cyclades.h>
|           ^~~~~~~~~~~~~~~~~~
| compilation terminated.
| make[2]: *** [Makefile:541: sanitizer_platform_limits_posix.lo] Error 1
| make[2]: Leaving directory '/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/gcc-8.3.0/build.aarch64-oe4t-linux.aarch64-oe4t-linux/aarch64-oe4t-linux/libsanitizer/sanitizer_common'
| make[1]: *** [Makefile:472: all-recursive] Error 1
| make[1]: Leaving directory '/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/gcc-8.3.0/build.aarch64-oe4t-linux.aarch64-oe4t-linux/aarch64-oe4t-linux/libsanitizer'
| make: *** [Makefile:342: all] Error 2
| ERROR: oe_runmake failed
| WARNING: /home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/run.do_compile.3850804:193 exit 1 from 'exit 1'
| WARNING: Backtrace (BB generated script):
| 	OE4T/tegra-demo-distro#1: bbfatal_log, /home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/run.do_compile.3850804, line 193
| 	OE4T/tegra-demo-distro#2: die, /home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/run.do_compile.3850804, line 174
| 	OE4T/tegra-demo-distro#3: oe_runmake, /home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/run.do_compile.3850804, line 163
| 	OE4T/tegra-demo-distro#4: do_compile, /home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/run.do_compile.3850804, line 158
| 	OE4T/tegra-demo-distro#5: main, /home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/run.do_compile.3850804, line 197
| ERROR: Execution of '/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/run.do_compile.3850804' failed with exit code 1:
| make "AR_FLAGS=" "CC_FOR_BUILD=gcc " "CFLAGS=-O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot-native= " "CXXFLAGS=-O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot-native=  -fvisibility-inlines-hidden" "CFLAGS_FOR_BUILD=-isystem/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot-native/usr/include -O2 -pipe" "CFLAGS_FOR_TARGET=" "INSTALL=/home/dingo/tegra-demo-distro/build/tmp/hosttools/install -c" "INSTALL_DATA=/home/dingo/tegra-demo-distro/build/tmp/hosttools/install -c -m 644" "INSTALL_PROGRAM=/home/dingo/tegra-demo-distro/build/tmp/hosttools/install -c" "INSTALL_SCRIPT=/home/dingo/tegra-demo-distro/build/tmp/hosttools/install -c" "JC1FLAGS=" "LDFLAGS=-Wl,-O1 -Wl,--hash-style=gnu -Wl,--as-needed -Wl,-z,relro,-z,now" "LIBCFLAGS=" "LIBCFLAGS_FOR_TARGET=" "MAKE=make" "MAKEINFO=/bin/sh /home/dingo/tegra-demo-distro/build/tmp/work-shared/gcc-8.3.0-r0/gcc-8.3.0/missing --run makeinfo " "PICFLAG=" "PICFLAG_FOR_TARGET=" "SHELL=/bin/sh" "RUNTESTFLAGS=" "exec_prefix=/usr" "infodir=/usr/share/info" "libdir=/usr/lib" "prefix=/usr" "includedir=/usr/include" "AR=aarch64-oe4t-linux-gcc-ar" "AS=aarch64-oe4t-linux-as " "LD=aarch64-oe4t-linux-ld --sysroot=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot " "LIBCFLAGS=" "NM=aarch64-oe4t-linux-gcc-nm" "PICFLAG=" "RANLIB=aarch64-oe4t-linux-gcc-ranlib" "DESTDIR=" all-recursive
| make[1]: Entering directory '/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/gcc-8.3.0/build.aarch64-oe4t-linux.aarch64-oe4t-linux/aarch64-oe4t-linux/libsanitizer'
| Making all in sanitizer_common
| make[2]: Entering directory '/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/gcc-8.3.0/build.aarch64-oe4t-linux.aarch64-oe4t-linux/aarch64-oe4t-linux/libsanitizer/sanitizer_common'
| /bin/sh ../libtool --tag=CXX   --mode=compile aarch64-oe4t-linux-g++  -march=armv8-a+crc -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security -Wdate-time --sysroot=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot -D_GNU_SOURCE -D_DEBUG -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS  -DHAVE_RPC_XDR_H=0 -DHAVE_TIRPC_RPC_XDR_H=0 -I. -I../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/sanitizer_common -I..  -I ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/include -isystem ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/include/system  -Wall -W -Wno-unused-parameter -Wwrite-strings -pedantic -Wno-long-long -fPIC -fno-builtin -fno-exceptions -fno-rtti -fomit-frame-pointer -funwind-tables -fvisibility=hidden -Wno-variadic-macros  -std=gnu++11  -DSANITIZER_LIBBACKTRACE -DSANITIZER_CP_DEMANGLE -I ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/../libbacktrace -I ../libbacktrace -I ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/../include -include ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/libbacktrace/backtrace-rename.h -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot-native=  -fvisibility-inlines-hidden -c -o sanitizer_platform_limits_posix.lo ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc
| libtool: compile:  aarch64-oe4t-linux-g++ -march=armv8-a+crc -fstack-protector-strong -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security -Wdate-time --sysroot=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot -D_GNU_SOURCE -D_DEBUG -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -DHAVE_RPC_XDR_H=0 -DHAVE_TIRPC_RPC_XDR_H=0 -I. -I../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/sanitizer_common -I.. -I ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/include -isystem ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/include/system -Wall -W -Wno-unused-parameter -Wwrite-strings -pedantic -Wno-long-long -fPIC -fno-builtin -fno-exceptions -fno-rtti -fomit-frame-pointer -funwind-tables -fvisibility=hidden -Wno-variadic-macros -std=gnu++11 -DSANITIZER_LIBBACKTRACE -DSANITIZER_CP_DEMANGLE -I ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/../libbacktrace -I ../libbacktrace -I ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/../include -include ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/libbacktrace/backtrace-rename.h -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0 -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0=/usr/src/debug/gcc-sanitizers/8.3.0-r0 -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot= -fdebug-prefix-map=/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/recipe-sysroot-native= -fvisibility-inlines-hidden -c ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc  -fPIC -DPIC -o .libs/sanitizer_platform_limits_posix.o
| ../../../../../../../../../work-shared/gcc-8.3.0-r0/gcc-8.3.0/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc:160:10: fatal error: linux/cyclades.h: No such file or directory
|  #include <linux/cyclades.h>
|           ^~~~~~~~~~~~~~~~~~
| compilation terminated.
| make[2]: *** [Makefile:541: sanitizer_platform_limits_posix.lo] Error 1
| make[2]: Leaving directory '/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/gcc-8.3.0/build.aarch64-oe4t-linux.aarch64-oe4t-linux/aarch64-oe4t-linux/libsanitizer/sanitizer_common'
| make[1]: *** [Makefile:472: all-recursive] Error 1
| make[1]: Leaving directory '/home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/gcc-8.3.0/build.aarch64-oe4t-linux.aarch64-oe4t-linux/aarch64-oe4t-linux/libsanitizer'
| make: *** [Makefile:342: all] Error 2
| WARNING: /home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/run.do_compile.3850804:193 exit 1 from 'exit 1'
| WARNING: Backtrace (BB generated script):
| 	OE4T/tegra-demo-distro#1: bbfatal_log, /home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/run.do_compile.3850804, line 193
| 	OE4T/tegra-demo-distro#2: die, /home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/run.do_compile.3850804, line 174
| 	OE4T/tegra-demo-distro#3: oe_runmake, /home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/run.do_compile.3850804, line 163
| 	OE4T/tegra-demo-distro#4: do_compile, /home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/run.do_compile.3850804, line 158
| 	OE4T/tegra-demo-distro#5: main, /home/dingo/tegra-demo-distro/build/tmp/work/armv8a-oe4t-linux/gcc-sanitizers/8.3.0-r0/temp/run.do_compile.3850804, line 197
| 
| Backtrace (metadata-relative locations):
| 	OE4T/tegra-demo-distro#1: bbfatal_log, /home/dingo/tegra-demo-distro/layers/meta/classes/logging.bbclass, line 72
| 	OE4T/tegra-demo-distro#2: die, /home/dingo/tegra-demo-distro/layers/meta/classes/base.bbclass, line 56
| 	OE4T/tegra-demo-distro#3: oe_runmake, /home/dingo/tegra-demo-distro/layers/meta/classes/base.bbclass, line 65
| 	OE4T/tegra-demo-distro#4: do_compile, /home/dingo/tegra-demo-distro/layers/meta-tegra/contrib/recipes-devtools/gcc-8/gcc-sanitizers.inc, line 32
ERROR: Task (/home/dingo/tegra-demo-distro/layers/meta-tegra/contrib/recipes-devtools/gcc-8/gcc-sanitizers_8.3.bb:do_compile) failed with exit code '1'
NOTE: Tasks Summary: Attempted 4973 tasks of which 4962 didn't need to be rerun and 1 failed.

Summary: 1 task failed:
  /home/dingo/tegra-demo-distro/layers/meta-tegra/contrib/recipes-devtools/gcc-8/gcc-sanitizers_8.3.bb:do_compile

--- a/contrib/recipes-devtools/gcc-8/gcc-8.3.inc
+++ b/contrib/recipes-devtools/gcc-8/gcc-8.3.inc
@@ -74,6 +74,7 @@ SRC_URI = "\
            file://0041-Add-a-recursion-limit-to-libiberty-s-demangling-code.patch \
            file://0042-PR-debug-86964.patch \
            file://0043-PR85434-Prevent-spilling-of-stack-protector-guard-s-.patch \
+           file://0044-libsanitizer-remove-cyclades.patch \
            file://CVE-2019-14250.patch \
            file://CVE-2019-15847_p1.patch \
            file://CVE-2019-15847_p2.patch \

--- a/contrib/recipes-devtools/gcc-8/gcc-8.3/0044-libsanitizer-remove-cyclades.patch
+++ b/contrib/recipes-devtools/gcc-8/gcc-8.3/0044-libsanitizer-remove-cyclades.patch
@@ -1,0 +1,121 @@
+From 2b40941d23b1570cdd90083b58fa0f66aa58c86e Mon Sep 17 00:00:00 2001
+From: Tamar Christina <tamar.christina@arm.com>
+Date: Fri, 21 May 2021 12:16:56 +0100
+Subject: [PATCH] libsanitizer: Remove cyclades from libsanitizer
+
+The Linux kernel has removed the interface to cyclades from
+the latest kernel headers[1] due to them being orphaned for the
+past 13 years.
+
+libsanitizer uses this header when compiling against glibc, but
+glibcs itself doesn't seem to have any references to cyclades.
+
+Further more it seems that the driver is broken in the kernel and
+the firmware doesn't seem to be available anymore.
+
+As such since this is breaking the build of libsanitizer (and so the
+GCC bootstrap[2]) I propose to remove this.
+
+[1] https://lkml.org/lkml/2021/3/2/153
+[2] https://gcc.gnu.org/bugzilla/show_bug.cgi?id=100379
+
+libsanitizer/ChangeLog:
+
+	PR sanitizer/100379
+	* sanitizer_common/sanitizer_common_interceptors_ioctl.inc: Cherry-pick
+	llvm-project revision f7c5351552387bd43f6ca3631016d7f0dfe0f135.
+	* sanitizer_common/sanitizer_platform_limits_posix.cc: Likewise.
+	* sanitizer_common/sanitizer_platform_limits_posix.h: Likewise.
+---
+ .../sanitizer_common_interceptors_ioctl.inc           |  9 ---------
+ .../sanitizer_platform_limits_posix.cc                | 11 -----------
+ .../sanitizer_platform_limits_posix.h                 | 10 ----------
+ 3 files changed, 30 deletions(-)
+
+diff --git a/libsanitizer/sanitizer_common/sanitizer_common_interceptors_ioctl.inc b/libsanitizer/sanitizer_common/sanitizer_common_interceptors_ioctl.inc
+index 5408ea17c59..7a9cd3f5968 100644
+--- a/libsanitizer/sanitizer_common/sanitizer_common_interceptors_ioctl.inc
++++ b/libsanitizer/sanitizer_common/sanitizer_common_interceptors_ioctl.inc
+@@ -365,15 +365,6 @@ static void ioctl_table_fill() {
+ 
+ #if SANITIZER_LINUX && !SANITIZER_ANDROID
+   // _(SIOCDEVPLIP, WRITE, struct_ifreq_sz); // the same as EQL_ENSLAVE
+-  _(CYGETDEFTHRESH, WRITE, sizeof(int));
+-  _(CYGETDEFTIMEOUT, WRITE, sizeof(int));
+-  _(CYGETMON, WRITE, struct_cyclades_monitor_sz);
+-  _(CYGETTHRESH, WRITE, sizeof(int));
+-  _(CYGETTIMEOUT, WRITE, sizeof(int));
+-  _(CYSETDEFTHRESH, NONE, 0);
+-  _(CYSETDEFTIMEOUT, NONE, 0);
+-  _(CYSETTHRESH, NONE, 0);
+-  _(CYSETTIMEOUT, NONE, 0);
+   _(EQL_EMANCIPATE, WRITE, struct_ifreq_sz);
+   _(EQL_ENSLAVE, WRITE, struct_ifreq_sz);
+   _(EQL_GETMASTRCFG, WRITE, struct_ifreq_sz);
+diff --git a/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc b/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc
+index d823a12190c..e8fce8a0287 100644
+--- a/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc
++++ b/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc
+@@ -157,7 +157,6 @@ typedef struct user_fpregs elf_fpregset_t;
+ # include <sys/procfs.h>
+ #endif
+ #include <sys/user.h>
+-#include <linux/cyclades.h>
+ #include <linux/if_eql.h>
+ #include <linux/if_plip.h>
+ #include <linux/lp.h>
+@@ -466,7 +465,6 @@ unsigned struct_ElfW_Phdr_sz = sizeof(Elf_Phdr);
+ 
+ #if SANITIZER_LINUX && !SANITIZER_ANDROID
+   unsigned struct_ax25_parms_struct_sz = sizeof(struct ax25_parms_struct);
+-  unsigned struct_cyclades_monitor_sz = sizeof(struct cyclades_monitor);
+ #if EV_VERSION > (0x010000)
+   unsigned struct_input_keymap_entry_sz = sizeof(struct input_keymap_entry);
+ #else
+@@ -833,15 +831,6 @@ unsigned struct_ElfW_Phdr_sz = sizeof(Elf_Phdr);
+ #endif // SANITIZER_LINUX || SANITIZER_FREEBSD
+ 
+ #if SANITIZER_LINUX && !SANITIZER_ANDROID
+-  unsigned IOCTL_CYGETDEFTHRESH = CYGETDEFTHRESH;
+-  unsigned IOCTL_CYGETDEFTIMEOUT = CYGETDEFTIMEOUT;
+-  unsigned IOCTL_CYGETMON = CYGETMON;
+-  unsigned IOCTL_CYGETTHRESH = CYGETTHRESH;
+-  unsigned IOCTL_CYGETTIMEOUT = CYGETTIMEOUT;
+-  unsigned IOCTL_CYSETDEFTHRESH = CYSETDEFTHRESH;
+-  unsigned IOCTL_CYSETDEFTIMEOUT = CYSETDEFTIMEOUT;
+-  unsigned IOCTL_CYSETTHRESH = CYSETTHRESH;
+-  unsigned IOCTL_CYSETTIMEOUT = CYSETTIMEOUT;
+   unsigned IOCTL_EQL_EMANCIPATE = EQL_EMANCIPATE;
+   unsigned IOCTL_EQL_ENSLAVE = EQL_ENSLAVE;
+   unsigned IOCTL_EQL_GETMASTRCFG = EQL_GETMASTRCFG;
+diff --git a/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h b/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h
+index 6a673a7c995..f921bf2b5b5 100644
+--- a/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h
++++ b/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h
+@@ -1040,7 +1040,6 @@ struct __sanitizer_cookie_io_functions_t {
+ 
+ #if SANITIZER_LINUX && !SANITIZER_ANDROID
+   extern unsigned struct_ax25_parms_struct_sz;
+-  extern unsigned struct_cyclades_monitor_sz;
+   extern unsigned struct_input_keymap_entry_sz;
+   extern unsigned struct_ipx_config_data_sz;
+   extern unsigned struct_kbdiacrs_sz;
+@@ -1385,15 +1384,6 @@ struct __sanitizer_cookie_io_functions_t {
+ #endif  // SANITIZER_LINUX || SANITIZER_FREEBSD
+ 
+ #if SANITIZER_LINUX && !SANITIZER_ANDROID
+-  extern unsigned IOCTL_CYGETDEFTHRESH;
+-  extern unsigned IOCTL_CYGETDEFTIMEOUT;
+-  extern unsigned IOCTL_CYGETMON;
+-  extern unsigned IOCTL_CYGETTHRESH;
+-  extern unsigned IOCTL_CYGETTIMEOUT;
+-  extern unsigned IOCTL_CYSETDEFTHRESH;
+-  extern unsigned IOCTL_CYSETDEFTIMEOUT;
+-  extern unsigned IOCTL_CYSETTHRESH;
+-  extern unsigned IOCTL_CYSETTIMEOUT;
+   extern unsigned IOCTL_EQL_EMANCIPATE;
+   extern unsigned IOCTL_EQL_ENSLAVE;
+   extern unsigned IOCTL_EQL_GETMASTRCFG;
+-- 
+2.27.0
+

--- a/external/virtualization-layer/recipes-containers/libnvidia-container-tools/libnvidia-container-tools-0.9.0/0005-Update-makefile-for-statically-linking-external-libt.patch
+++ b/external/virtualization-layer/recipes-containers/libnvidia-container-tools/libnvidia-container-tools-0.9.0/0005-Update-makefile-for-statically-linking-external-libt.patch
@@ -1,0 +1,30 @@
+From b025626b42f892bfe1f3683531a9031fec4f6834 Mon Sep 17 00:00:00 2001
+From: Matt Madison <matt@madison.systems>
+Date: Fri, 23 Jul 2021 11:04:56 -0700
+Subject: [PATCH] Update makefile for statically linking external libtirpc
+
+Signed-off-by: Matt Madison <matt@madison.systems>
+---
+ Makefile | 6 +++---
+ 1 file changed, 3 insertions(+), 3 deletions(-)
+
+diff --git a/Makefile b/Makefile
+index cd44c73..6f24b80 100644
+--- a/Makefile
++++ b/Makefile
+@@ -127,9 +127,9 @@ LDLIBS   := $(LDLIBS)
+ # Library flags (recursively expanded to handle target-specific flags)
+ LIB_CPPFLAGS       = -DNV_LINUX -isystem $(DEPS_DIR)$(includedir) -include $(BUILD_DEFS)
+ LIB_CFLAGS         = -fPIC
+-LIB_LDFLAGS        = -L$(DEPS_DIR)$(libdir) -shared -Wl,-soname=$(LIB_SONAME)
+-LIB_LDLIBS_STATIC  = -l:libnvidia-modprobe-utils.a
+-LIB_LDLIBS_SHARED  = -ldl -lcap -ltirpc
++LIB_LDFLAGS        = -shared -Wl,-soname=$(LIB_SONAME)
++LIB_LDLIBS_STATIC  = -L$(DEPS_DIR)$(libdir) -l:libnvidia-modprobe-utils.a -L=$(libdir) -l:libtirpc.a
++LIB_LDLIBS_SHARED  = -ldl -lcap -lpthread
+ ifeq ($(WITH_LIBELF), yes)
+ LIB_CPPFLAGS       += -DWITH_LIBELF
+ LIB_LDLIBS_SHARED  += -lelf
+-- 
+2.30.2
+

--- a/external/virtualization-layer/recipes-containers/libnvidia-container-tools/libnvidia-container-tools_0.9.0.bb
+++ b/external/virtualization-layer/recipes-containers/libnvidia-container-tools/libnvidia-container-tools_0.9.0.bb
@@ -14,7 +14,7 @@ DEPENDS = " \
     pkgconfig-native \
     libcap \
     elfutils \
-    libtirpc \
+    libtirpc126 \
     ldconfig-native \
 "
 LICENSE = "BSD-3-Clause & GPLv3 & Proprietary"
@@ -37,6 +37,7 @@ SRC_URI = "git://github.com/NVIDIA/libnvidia-container.git;protocol=https;name=l
            file://0002-common.mk-Set-JETSON-variable-if-not-set-before.patch \
            file://0003-Fix-mapping-of-library-paths-for-jetson-mounts.patch \
            file://0004-Fix-build.h-generation-for-cross-builds.patch \
+           file://0005-Update-makefile-for-statically-linking-external-libt.patch \
            "
 
 SRC_URI[modprobe.md5sum] = "f82b649e7a0f1d1279264f9494e7cf43"
@@ -64,7 +65,7 @@ def build_date(d):
 # include bmake-native which does not exist at the moment.
 EXTRA_OEMAKE = "EXCLUDE_BUILD_FLAGS=1 PLATFORM=${HOST_ARCH} JETSON=TRUE WITH_LIBELF=yes ${@build_date(d)} ${PACKAGECONFIG_CONFARGS}"
 
-CFLAGS_prepend = " -I=/usr/include/tirpc "
+CFLAGS_prepend = " -I=/usr/include/tirpc-1.2.6 "
 
 export OBJCPY="${OBJCOPY}"
 

--- a/external/virtualization-layer/recipes-containers/libnvidia-container-tools/libtirpc126_1.2.6.bb
+++ b/external/virtualization-layer/recipes-containers/libnvidia-container-tools/libtirpc126_1.2.6.bb
@@ -1,0 +1,27 @@
+SUMMARY = "Transport-Independent RPC library"
+DESCRIPTION = "Libtirpc is a port of Suns Transport-Independent RPC library to Linux"
+SECTION = "libs/network"
+HOMEPAGE = "http://sourceforge.net/projects/libtirpc/"
+BUGTRACKER = "http://sourceforge.net/tracker/?group_id=183075&atid=903784"
+LICENSE = "BSD-3-Clause"
+LIC_FILES_CHKSUM = "file://COPYING;md5=f835cce8852481e4b2bbbdd23b5e47f3 \
+                    file://src/netname.c;beginline=1;endline=27;md5=f8a8cd2cb25ac5aa16767364fb0e3c24"
+
+SRC_URI = "${SOURCEFORGE_MIRROR}/libtirpc/libtirpc-${PV}.tar.bz2"
+SRC_URI[sha256sum] = "4278e9a5181d5af9cd7885322fdecebc444f9a3da87c526e7d47f7a12a37d1cc"
+
+S = "${WORKDIR}/libtirpc-${PV}"
+
+inherit autotools pkgconfig
+
+DISABLE_STATIC = ""
+EXTRA_OECONF = "--disable-gssapi --enable-static"
+
+do_install_append() {
+    rm -r ${D}${sysconfdir} ${D}${datadir} ${D}${libdir}/pkgconfig
+    rm ${D}${libdir}/*.so*
+    mv ${D}${includedir}/tirpc ${D}${includedir}/tirpc-1.2.6
+}
+
+inherit nopackages
+

